library/core/src/cmp.rs

```diff
@@ -1476,11 +1476,11 @@ mod impls {
             impl const Ord for $t {
                 #[inline]
                 fn cmp(&self, other: &$t) -> Ordering {
-                    // The order here is important to generate more optimal assembly.
-                    // See <https://github.com/rust-lang/rust/issues/63758> for more info.
-                    if *self < *other { Less }
-                    else if *self == *other { Equal }
-                    else { Greater }
+                    let mut res = 0i8;
+                    res -= (*self < *other) as i8;
+                    res += (*self > *other) as i8;
+                    // SAFETY: The discriminants of Ord were chosen to permit this
```

Comment:
I believe this should say "Ordering" rather than "Ord".

Should there be a comment [near the discriminant values](https://doc.rust-lang.org/src/core/cmp.rs.html#3360) mentioning that safety of some impls depends on the values not being changed? Or a static assertion here that the discriminant values are the expected ones?

Reply:
The discriminants are [visible on stable](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=99977f2338782b141f6be506a3464c83) and it's explicitly `#[repr(i8)]`, so we probably couldn't change them anyway.  But noting that in the enum's documentation -- either `///` or `//` -- seems quite reasonable.